### PR TITLE
Replace sorbet-runtime-stub with sorbet-runtime

### LINKED ISSUE #11521: sorbet-runtime-stub is deprecated
> **DEPRECATED: This gem should not be used anymore.**
>
> As of June 2021, `sorbet-runtime-stub` is no more maintained and no further versions will be published.
>
> We decided to stop supporting this gem for multiple reasons:
> * It's not feature complete (missing T::Struct support and other Sorbet delicacies)
> * It's hard to keep up-to-date with the real sorbet-runtime
> * It provides a false sense of security for the users
>
> You should include sorbet-runtime and [disable the checks](https://sorbet.org/docs/runtime#changing-the-runtime-behavior) you don't want to apply in production env.
>
> If you're in the Shopify org, take a look at https://github.com/shopify/shopify-types.

### PR DESCRIPTION
**Pull request by Bo98**
Resolves #11521.

This does have more overhead than the stub but I'm not convinced it's noticeable. I've done a monkey patch disabling some more common parts of runtime checking though to be on the safer side.

If the overhead is not much, we could expxand this in the future by not doing these monkey patches and doing things such as a one-pass basic evaluation of `sig` blocks to allow for example [abstract classes](https://sorbet.org/docs/abstract) to work if we want to make use of them. We don't need to commit to full level runtime checking to do this - [there's different levels on the spectrum](https://sorbet.org/docs/runtime).

Should be able to replace #13563 if the user does `brew install-bundler-gems --groups=` first to clear previous settings. I would still however like information to fix the install error some people have since it'll still be an issue for `brew typecheck`. I cannot reproduce the issue myself.

### CODE REVIEW

**Comment by Bo98** [comment]
`brew install-bundler-gems` doesn't uninstall gems properly, which is causing `brew doctor` to fail in CI. I feel like we've had that issue before.

File: Library/Homebrew/standalone/sorbet.rb [hunk 1]
```diff
@@ -1,11 +1,41 @@
...
-def gem(name, *)
-  raise Gem::LoadError if name == "sorbet-runtime"
+require "sorbet-runtime"
 
-  super
-end
+# Disable runtime checking unless enabled.
+unless ENV["HOMEBREW_SORBET_RUNTIME"]
+  # @private
+  module TNoChecks
```
Comment (MikeMcQuaid) [inline Library/Homebrew/standalone/sorbet.rb:9]: I don't understand what any of the code in this file is doing and why it's here 😅. Could it get some more comments?

**Review by MikeMcQuaid — APPROVED** [review]
One comment about more comments but otherwise seems fine thanks 👍🏻

**Comment by MikeMcQuaid** [comment]
> Cop out fix is to roll it in with the next Sorbet version bump but we should probably fix it properly.

Yeh, that'd be good. 

> because `bundle check` only fails if a declared gem is not installed - not if an installed gem is not declared

Maybe needs an explicit `bundle cleanup` or `bundle install --cleanup` (neither are correct syntax, I think) to handle these cases?

File: Library/Homebrew/standalone/sorbet.rb (hunk 1 above)
Comment (Bo98) [reply Library/Homebrew/standalone/sorbet.rb:9]: > `# Disable runtime checking unless enabled.`

I could maybe add more?

Comment (MikeMcQuaid) [reply Library/Homebrew/standalone/sorbet.rb:9]: Yeh, please, pretty much just this comment but in Ruby 😁

**Comment by Bo98** [comment]
> Maybe needs an explicit `bundle cleanup` or `bundle install --cleanup` (neither are correct syntax, I think) to handle these cases?

`bundle cleanup` seems to just make #5798 resurface.

**Comment by Bo98** [comment]
Ah looks like Bundler might have finally fixed the extensions directory mess in the standalone `setup.rb` we have: https://github.com/rubygems/rubygems/commit/a9dae93d5d19b7e249fcb0f774f42c89e6976247. Though that will mean using a newer version of Bundler.

Not really relevant to this PR but good to know nonetheless.

**Comment by Bo98** [comment]
Pushed a Sorbet update here since I don't really know how to handle bundle cleanup without breaking everything.

**Review by MikeMcQuaid — APPROVED** [review]

**Pull request merged**